libbeat/cmd/instance/beat.go

```diff
@@ -457,8 +453,11 @@ func (b *Beat) Setup(settings Settings, bt beat.Creator, setup SetupSettings) er
 				}
 
 				// prepare index by loading templates, lifecycle policies and write aliases
-
-				m := b.index.Manager(esClient, idxmgmt.BeatsAssets(b.Fields))
+				fields, err := b.Mapping.GetBytes()
+				if err != nil {
+					return err
+				}
+				m := b.index.Manager(esClient, idxmgmt.BeatsAssets(fields))
```

Comment:
The 'BeatsAssets' has been a place-holder, due to not having some better options availble. I'd be in favor of passing a more rich interface instead of a raw blob to be parsed and interpreted somewhere else.

If possible we want to be able to install multiple templates in the future, each with it's own set of fields. A more generic type better being integrated with the index manager would be helpful.

The index manager is supposed to be the only component/interface Beats should use for configuring and preparing indices (facade pattern). Maybe we can move the fields/templating support inside the index manager?